src/main/java/com/mongodb/kafka/connect/source/schema/AvroSchema.java

```diff
@@ -218,11 +208,12 @@ static Schema createSchema(
                 .filter(s -> s.getType() != org.apache.avro.Schema.Type.NULL)
                 .findFirst();
         if (optionalSchema.isPresent()) {
-          return createSchema(optionalSchema.get(), true, context);
+          return createSchema(optionalSchema.get(), true, null, context);
         }
         throw new IllegalStateException();
       case NULL:
       case ENUM:
+      case FIXED:
```

Comment:
`validateJsonSchema` complains when encounters `FIXED`, so there is no reason to support converting `FIXED` in `createSchema`. We can also see that `NULL` and `ENUM` have been treated in the same way as `FIXED`.